client/lib/analytics/ad-tracking/record-order.js

```diff
@@ -29,9 +30,10 @@ import './setup';
  *
  * @param {Object} cart - cart as `ResponseCart` object
  * @param {number} orderId - the order id
+ * @param {import('@automattic/data-stores/src').SiteDetails} selectedSiteData - site details
  * @returns {void}
  */
-export async function recordOrder( cart, orderId ) {
+export async function recordOrder( cart, orderId, selectedSiteData ) {
```

Comment:
As mentioned below, seems like we can only pass the product slug, and not the entire site details object.